### ISSUE TITLE
GraphQL mock

### ISSUE THREAD

**Issue by zuffik** [mockoon/mockoon#143]
Any chance to get GraphQL mocking alongside REST API mocking?

**Comment by mtzfactory** [comment]
Would be amazing to have this feature!

**Comment by StarNeit** [comment]
Would be great to have graphql too!

**Comment by 255kb** [comment]
I started working on the GraphQL implementation. However, it raises a lot of questions regarding rules, statuses, etc. 
I would like to gather a bit of feedback from Mockoon's users on how they imagine this GraphQL feature. Feel free to share your views here or on the [Discord server](https://discord.gg/MutRpsY5gE) (you can directly DM me @255bk). 
Thank you!

**Comment by panchalkalpesh** [comment]
Until this mockoon supports GraphQL, I have a fallback on [graphql-faker](https://github.com/APIs-guru/graphql-faker)

**Comment by 255kb** [comment]
Thank you for the feedback @panchalkalpesh !

**Comment by 142934** [comment]
How to use GrapQL with mockoon?
actually I am trying to use it but as the endpoint of all apis is same so it is returning the response of first API all the time ,It is not able to differentiate the apis based on the GraphQl queries

**Comment by 255kb** [comment]
@142934 GraphQL is not supported yet.

**Comment by 142934** [comment]
> @142934 GraphQL is not supported yet.

When it is expected to come? could you please launch this feature ASAP?

**Comment by 255kb** [comment]
@142934 
I pondered for a while whether I should delete your comment or not. Instead, I will try to educate and move on with my day.
I've been working on Mockoon for the past seven years, during my free time, and for the last three years full-time without being paid because I believe in Mockoon's potential to be a leader in this market. I'm mostly alone in the team, though I do receive frequent help from fantastic open-source contributors and friends. The SaaS I'm building around this project is growing, but it is nowhere near covering my bills (it's also just the very beginning of it). To be honest, not many users (individuals or companies) are sponsoring the project despite numerous calls for support. So, I'm still freelancing on the side while trying to dedicate as much time as possible to Mockoon.
Here is where the beauty of open source shines. Open source is not only about releasing something for the world to use but also about random strangers contributing back to projects. I must say that in this regard, I have far more consumers than contributors. I don't blame anyone; we are all busy. I invite you to contribute to the projects you use, especially if a feature is vital to you. You might get a glimpse of what it takes to add such a significant feature to a project used by tens of thousands of users worldwide.
Finally, GraphQL is the perfect example of the kind of issue I dread:
I know nothing about GraphQL.
Massive changes are required in terms of UI and UX.
Extensive interaction with other features is necessary.
Nobody has tried to contribute even a bit by giving an opinion on how it should interact with other features (rules, etc.) or by trying to define the scope of an initial minimal implementation. Nothing.
It is certainly upvoted a lot, and having this feature would be awesome, but it is not closer to being implemented now than one hour ago.

**Comment by alexbuildaz** [comment]
Would be great if mockoon fallback to the proxy URL when the route rules doesn't match. The current issue for me is that when the rules don't match, it stills returns the mock because the route path matches (because of graphql), while I was expecting it to call the proxy.

**Comment by 255kb** [comment]
> Would be great if mockoon fallback to the proxy URL when the route rules doesn't match. The current issue for me is that when the rules don't match, it stills returns the mock because the route path matches (because of graphql), while I was expecting it to call the proxy.

This is not relly related to GraphQL. There is a ["fallback" mode](https://mockoon.com/docs/latest/route-responses/multiple-responses/#fallback-mode) that you can activate and could help you achieve what you want.

**Comment by maheshdagility** [comment]
Can we use Graphql in mockoon now

**Comment by mamatagelanee07** [comment]
@255kb Any plans for GraphQL support soon?

**Comment by 255kb** [comment]
> [@255kb](https://github.com/255kb) Any plans for GraphQL support soon?

No. It's on the roadmap but it's also a massive feature. We are a small team with limited resources, and more importantly, limited knowledge of GraphQL. We would rely on a community effort to add GraphQL, but it requires a really dedicated developer ready to commit for multiple months of work to see this through. We haven't met this person yet :)
